src/index.js

```diff
@@ -1,14 +1,19 @@
 import defaultErrorHandler from "./default-error-handler";
+import isPlainObject from "lodash.isplainobject";
 
 export default function reduxUnhandledAction(callback = defaultErrorHandler) {
-	return ({getState}) => (next) => (action) => {
-		const prevState = getState();
-		const result = next(action);
-		const nextState = getState();
-		if (prevState === nextState) {
-			callback(action);
-		}
-		return result;
-	};
-
+    return ({ getState }) => (next) => (action) => {
+        if (isPlainObject(action)) {
+            if (typeof action.type !== 'undefined') {
```

Comment:
instead of nested if can we just put this in one statement, also please use double quotes on undefined string. otherwise looks great. Thanks for test cases as well!
```js
if (isPlainObject(action) && typeof action.type !== "undefined") {
```